Bottom: 474266e0219aa3d67b3054345c4b24d3e5e1ecdc
Top:    70414e400d3a837f9784cba2ead735eb023f00e7
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2011-08-30 00:18:00 +0100

Refresh of draw-c-add-const-qualifiers-to

---

diff --git a/src/draw.c b/src/draw.c
index e0da9a9..410d15e 100644
--- a/src/draw.c
+++ b/src/draw.c
@@ -81,14 +81,14 @@ static bool doing_assy = false;
 /* ---------------------------------------------------------------------------
  * some local prototypes
  */
-static void DrawEverything (BoxTypePtr);
+static void DrawEverything (const BoxType *);
 static void DrawPPV (int group, const BoxType *);
 static void DrawLayerGroup (int, const BoxType *);
 static void AddPart (void *);
 static void SetPVColor (PinTypePtr, int);
 static void DrawEMark (ElementTypePtr, Coord, Coord, bool);
-static void DrawMask (int side, BoxType *);
-static void DrawPaste (int side, BoxType *);
+static void DrawMask (int side, const BoxType *);
+static void DrawPaste (int side, const BoxType *);
 static void DrawRats (BoxType *);
 static void DrawSilk (int side, const BoxType *);
 
@@ -623,7 +623,7 @@ PrintAssembly (int side, const BoxType * drawn_area)
  * initializes some identifiers for a new zoom factor and redraws whole screen
  */
 static void
-DrawEverything (BoxTypePtr drawn_area)
+DrawEverything (const BoxType *drawn_area)
 {
   int i, ngroups, side;
   int component, solder;
@@ -970,7 +970,7 @@ DrawSilk (int side, const BoxType * drawn_area)
 
 
 static void
-DrawMaskBoardArea (int mask_type, BoxType *screen)
+DrawMaskBoardArea (int mask_type, BoxType *drawn_area)
 {
   /* Skip the mask drawing if the GUI doesn't want this type */
   if ((mask_type == HID_MASK_BEFORE && !gui->poly_before) ||
@@ -979,18 +979,18 @@ DrawMaskBoardArea (int mask_type, BoxType *screen)
 
   gui->use_mask (mask_type);
   gui->set_color (Output.fgGC, PCB->MaskColor);
-  if (screen == NULL)
+  if (drawn_area == NULL)
     gui->fill_rect (Output.fgGC, 0, 0, PCB->MaxWidth, PCB->MaxHeight);
   else
-    gui->fill_rect (Output.fgGC, screen->X1, screen->Y1,
-                                 screen->X2, screen->Y2);
+    gui->fill_rect (Output.fgGC, drawn_area->X1, drawn_area->Y1,
+                                 drawn_area->X2, drawn_area->Y2);
 }
 
 /* ---------------------------------------------------------------------------
  * draws solder mask layer - this will cover nearly everything
  */
 static void
-DrawMask (int side, BoxType * screen)
+DrawMask (int side, const BoxType *screen)
 {
   int thin = TEST_FLAG(THINDRAWFLAG, PCB) || TEST_FLAG(THINDRAWPOLYFLAG, PCB);
 
@@ -1019,7 +1019,7 @@ DrawMask (int side, BoxType * screen)
  * draws solder paste layer for a given side of the board
  */
 static void
-DrawPaste (int side, BoxType *drawn_area)
+DrawPaste (int side, const BoxType *drawn_area)
 {
   gui->set_color (Output.fgGC, PCB->ElementColor);
   ALLPAD_LOOP (PCB->Data);
